packages/medusa/src/services/cart.ts

```diff
@@ -1778,6 +1778,21 @@ class CartService extends TransactionBaseService {
                 return this.paymentProviderService_
                   .withTransaction(transactionManager)
                   .updateSession(paymentSession, paymentSessionInput)
+                  .catch(async (e) => {
+                    if (!paymentSession.is_selected) {
+                      this.logger_.warn(
+                        `Unable to update the payment session for ${paymentSession.provider_id}. This session will be deleted since it is not selected${EOL}${e.stack}`
+                      )
+                      return await this.paymentProviderService_.deleteSession(
+                        paymentSession
+                      )
+                    }
```

Comment:
praise: I believe this is the best way of handling errors caused by third-party providers. Does not block the checkout flow for the end customer and allows the server admin to pick up the issue.

What do you think @srindom?